### ISSUE TITLE
Exception in plugin .ignore. Occured once since the last clear. Unread

### ISSUE THREAD

**Issue by djzara** [hsz/idea-gitignore#381]
I received the following error after some updates performed on my system ruby installation. Please advise as to whether or not this is a plugin issue, or an underlying system issue with either RubyMine or Jetbrains' platform

```
Invalid root file:///usr/local/lib/ruby/vendor_ruby in untitled -> < ruby-2.4.1-p111 >
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at com.intellij.openapi.roots.impl.RootIndex.a(RootIndex.java:195)
	at com.intellij.openapi.roots.impl.RootIndex.a(RootIndex.java:134)
	at com.intellij.openapi.roots.impl.RootIndex.<init>(RootIndex.java:68)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.b(DirectoryIndexImpl.java:111)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getInfoForFile(DirectoryIndexImpl.java:146)
	at com.intellij.openapi.roots.impl.FileIndexBase.getInfoForFileOrDirectory(FileIndexBase.java:57)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.getModuleForFile(ProjectFileIndexImpl.java:108)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.getModuleForFile(ProjectFileIndexImpl.java:101)
	at com.intellij.openapi.module.ModuleUtilCore.findModuleForFile(ModuleUtilCore.java:74)
	at com.intellij.openapi.vcs.impl.ModuleDefaultVcsRootPolicy.getMatchContext(ModuleDefaultVcsRootPolicy.java:111)
	at com.intellij.openapi.vcs.impl.projectlevelman.NewMappings.getMappingFor(NewMappings.java:215)
	at com.intellij.openapi.vcs.impl.projectlevelman.NewMappings.getVcsFor(NewMappings.java:243)
	at com.intellij.openapi.vcs.impl.ProjectLevelVcsManagerImpl.getVcsFor(ProjectLevelVcsManagerImpl.java:278)
	at com.intellij.openapi.vcs.impl.VcsFileStatusProvider.b(VcsFileStatusProvider.java:183)
	at com.intellij.openapi.vcs.impl.VcsFileStatusProvider.isSupported(VcsFileStatusProvider.java:179)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager.a(LineStatusTrackerManager.java:265)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager.a(LineStatusTrackerManager.java:244)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager.a(LineStatusTrackerManager.java:238)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager.b(LineStatusTrackerManager.java:217)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager.access$1600(LineStatusTrackerManager.java:67)
	at com.intellij.openapi.vcs.impl.LineStatusTrackerManager$MyFileStatusListener.fileStatusesChanged(LineStatusTrackerManager.java:419)
	at com.intellij.openapi.vcs.impl.FileStatusManagerImpl.fileStatusesChanged(FileStatusManagerImpl.java:216)
	at mobi.hsz.idea.gitignore.util.CacheMap.refresh(CacheMap.java:194)
	at mobi.hsz.idea.gitignore.util.CacheMap.cleanup(CacheMap.java:180)
	at mobi.hsz.idea.gitignore.IgnoreManager$1.fileDeleted(IgnoreManager.java:171)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.EventDispatcher.dispatch(EventDispatcher.java:99)
	at com.intellij.util.EventDispatcher.access$200(EventDispatcher.java:35)
	at com.intellij.util.EventDispatcher$2.invoke(EventDispatcher.java:79)
	at com.sun.proxy.$Proxy17.fileDeleted(Unknown Source)
	at com.intellij.openapi.vfs.impl.BulkVirtualFileListenerAdapter.fireAfter(BulkVirtualFileListenerAdapter.java:85)
	at com.intellij.openapi.vfs.impl.BulkVirtualFileListenerAdapter.after(BulkVirtualFileListenerAdapter.java:56)
	at sun.reflect.GeneratedMethodAccessor59.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:117)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:437)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:398)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:387)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:373)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:44)
	at com.intellij.util.messages.impl.MessageBusImpl$2.invoke(MessageBusImpl.java:230)
	at com.sun.proxy.$Proxy18.after(Unknown Source)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.processEvents(PersistentFSImpl.java:779)
	at com.intellij.openapi.vfs.newvfs.RefreshSessionImpl.c(RefreshSessionImpl.java:196)
	at com.intellij.openapi.application.WriteAction.run(WriteAction.java:91)
	at com.intellij.openapi.vfs.newvfs.RefreshSessionImpl.fireEvents(RefreshSessionImpl.java:183)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:86)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:109)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:118)
	at com.intellij.openapi.vfs.newvfs.RefreshQueueImpl.a(RefreshQueueImpl.java:81)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:303)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:410)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:399)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:827)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:655)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by hsz** [comment]
@djzara Thank you for this report. This issue was resolved mostly by Migration to the native IDE indexing #367
It will be released with v2.0.0 as soon as other issues will be resolved.

**Comment by djzara** [comment]
Thank you for the response! I'm more than willing to be patient.
